package.json

```
@@ -78,7 +78,7 @@
   "scripts": {
     "lint": "yarn eslint --max-warnings=0 --cache --fix \"**/*.{js,ts,jsx,tsx}\"",
     "lint:markdown": "markdownlint --config linters/.markdownlint.json README.md */README.md",
-    "postinstall": "is-ci || husky install",
+    "postinstall": "is-ci || (command -v husky >/dev/null 2>&1 && husky init || echo 'Husky is not installed, skipping husky init.')",
```

@mckelveygreg this postinstall script is what is running on downstream repos when they install the @20i/eslint package.

> Husky should only be running when running this specific project, and none of that should leak to any downstream repos. That's not happening right? If so, then I screwed something up when publishing to npm!!
>
> What would be a good reason to not have husky running locally?

---

https://github.com/typicode/husky/issues/884

I only encountered it as I use a lot of greenfield / virtual environments to diagnose undocumented dependency issues. Helps to catch items that re not explicitly present in a project's configuration/documentation. So if I have husky installed globally or already in the project, then this step would silently pass as it was.

---

It looks like the pinst --disable did not execute on the publish, as that would remove that portion, right?

---

```suggestion
    "postinstall": "is-ci || husky init",
```
So when we sort out the publishing, I guess this change can be simplified.

---

Oh, sorry to lead astray, looks like the postinstall removal is fixed in current versions.

And I need to publish a patch to the 2.4.0 version, that's the one that was published with the postinstall script present.

---

Hmm, npm seems to think that one is set up correctly, though? https://www.npmjs.com/package/@20i/eslint-config/v/2.4.0?activeTab=code